tests/testthat/test-shinytest2-tm_data_table.R

```diff
@@ -0,0 +1,39 @@
+test_that("e2e: tm_front_page initializes without errors", {
+  skip_if_too_deep(5)
+  app_driver <- app_driver_tm_data_table()
+
+  app_driver$expect_no_shiny_error()
+  app_driver$stop()
+})
+
+test_that("e2e: tm_front_page displays data table", {
```

Comment:
I can use the expect_match() function to test the content for partial matches, although I'm not completely convinced useful here. I was checking caption as its passed paramter to module.

Reply:
we can extract the data frame by code `data.frame(rvest::html_table(app_driver$get_html_rvest(app_driver$active_module_element("iris-data_table")))[[2]]`
The table includes character to factor conversions and features pagination so do not show complete table. Should we instead focus on verifying the presence of the table rather than its specific content?

Reply:
It the current matching test only provides partial matches and does not always ensure correct output. This may cause issues, especially with repetitive values. 

![image](https://github.com/insightsengineering/teal.modules.general/assets/6700955/ab3a199a-06f9-4537-bc22-31cf03fd99ec)

Reply:
I think the test is missing some specs : P sorry
```r
  app_driver$set_active_module_input("method", "Z-score")
  app_driver$set_active_module_input("zscore_slider", 1.5)
 ```

Reply:
Yeah, I don't remember what values were set for this test so I updated the whole test : D
774ed6d9059409d0e4afbab8efa6ae44db7febe5

Reply:
Here is test for comparing displayed table value. [fa3e229](https://github.com/insightsengineering/teal.modules.general/pull/719/commits/fa3e22905fc1162d03c81bb3850d239f1d02f4a1)